pkg/catalog/repeater.go

```diff
@@ -21,13 +21,13 @@ func (mc *MeshCatalog) repeater() {
 			if chosenIdx, message, ok := reflect.Select(cases); ok {
 				if ann, ok := message.Interface().(announcements.Announcement); ok {
 					mc.handleAnnouncement(ann)
-				}
```

Comment:
by moving the `broadcast` call inside the assertion, we ensure that proxies are updated only when the events arriving here are of type `Announcement`